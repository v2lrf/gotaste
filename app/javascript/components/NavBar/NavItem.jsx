import React from 'react'
import PropTypes from 'prop-types'

export function NavItem({ to, label }) {
  return (
    <li className="mr-6">
      <a
        className="block text-white no-underline hover:underline focus:underline"
        href={to}
      >
        {label}
      </a>
    </li>
  )
}

NavItem.propTypes = {
  to: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
}

export default NavItem
