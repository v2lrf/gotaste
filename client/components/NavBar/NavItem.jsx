import React from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import { withRouter } from 'next/router'
import classNames from 'classnames'

function NavItem({ router, to, children, asActive }) {
  const linkClasses = classNames(
    'block py-4 px-4 sm:py-1 sm:px-2 text-red-light font-semibold no-underline hover:bg-red-lightest rounded',
    {
      'bg-red-lightest': router.pathname === to && asActive
    }
  )
  return (
    <li>
      <Link prefetch href={to}>
        <a className={linkClasses}>{children}</a>
      </Link>
    </li>
  )
}

NavItem.propTypes = {
  // Eslint disabled because it's a prop provided by Next.js
  // eslint-disable-next-line
  router: PropTypes.object,
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  asActive: PropTypes.bool
}

NavItem.defaultProps = {
  asActive: false
}

export default withRouter(NavItem)
