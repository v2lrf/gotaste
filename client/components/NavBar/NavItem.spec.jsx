import React from 'react'
import NavItem from './NavItem'

describe('NavItem', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      to: '/some-path'
    }
    component = shallow(<NavItem {...props}>Some content</NavItem>)
  })

  describe('Snapshots', () => {
    it('should match the default snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('should match the active snapshot', () => {
      component.setProps({
        asActive: true,
        router: {
          pathname: '/some-path'
        }
      })
      expect(component).toMatchSnapshot()
    })
  })
})
