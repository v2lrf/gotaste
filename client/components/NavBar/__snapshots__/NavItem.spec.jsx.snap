// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Snapshots should match the active snapshot 1`] = `
<NavItem
  asActive={true}
  router={
    Object {
      "pathname": "/some-path",
    }
  }
  to="/some-path"
>
  Some content
</NavItem>
`;

exports[`NavItem Snapshots should match the default snapshot 1`] = `
<NavItem
  asActive={false}
  to="/some-path"
>
  Some content
</NavItem>
`;
